Px.Utils/Validation/SyntaxValidation/SyntaxValidation.cs

```diff
@@ -0,0 +1,341 @@
+        /// <param name="syntaxConf">Syntax configuration object</param>
+        /// <param name="report">Report object that stores any issues that were ran into during validation process</param>
+        /// <param name="filename">Name of the PX file being validated</param>
+        /// <returns>Character encoding if found</returns>
+        public static async Task<Encoding?> GetEncodingAsync(Stream stream, PxFileSyntaxConf syntaxConf)
+        {
+            try
+            {
+                return await PxFileMetadataReader.GetEncodingAsync(stream, syntaxConf);
+            }
+            catch (InvalidPxFileMetadataException)
+            {
+                return null;
+            }
+        }
+
+        ///<summary>
+        /// Validates a collection of objects using a collection of validation functions and adds any feedback to the provided validationFeedback.
+        ///</summary>
+        /// <param name="objects">The collection of <see cref="ValidationObject"/> objects to be validated.</param>
+        /// <param name="validationFunctions">The collection of <see cref="IValidationFunction"/> validation functions to be used for validation.</param>
+        /// <param name="report">The <see cref="ValidationReport"/> validationFeedback where validation feedback will be added.</param>
+        /// <param name="syntaxConf">The syntax configuration for the PX file.</param>
+        public static void ValidateObjects(IEnumerable<ValidationObject> objects, IEnumerable<ValidationFunctionDelegate> validationFunctions, List<ValidationFeedbackItem> validationFeedback, PxFileSyntaxConf syntaxConf)
+        {
+            foreach (var @object in objects)
+            {
+                foreach (ValidationFunctionDelegate function in validationFunctions)
+                {
+                    ValidationFeedbackItem? feedback = function(@object, syntaxConf);
+                    if (feedback is not null)
+                    {
+                        validationFeedback.Add((ValidationFeedbackItem)feedback);
+                    }
+                }
+            }
+        }
+
+        private static List<ValidationKeyValuePair> BuildKeyValuePairs(List<ValidationEntry> validationEntries, PxFileSyntaxConf syntaxConf)
+        {
+            return validationEntries.Select(entry =>
+            {
+                // Split the entry string into a key and a value from the first valid keyword separator
+                int[] keywordSeparatorIndeces = SyntaxValidationUtilityMethods.FindKeywordSeparatorIndeces(entry.EntryString, syntaxConf);
+
+                // Key is the part of the entry string before the first keyword separator index
+                string key = entry.EntryString[..keywordSeparatorIndeces[0]];
+                // Remove preceding linebreaks and carriage returns from the key
+                key = key
+                    .TrimStart(syntaxConf.Symbols.CarriageReturn)
```

Comment:
These can be merged into a single TrimStart() call. It has an array parameter overload.